src/Groups/matrices/form_group.jl

```diff
@@ -791,20 +882,7 @@ function _isometry_group_via_decomposition(L::ZZLat; closed = true, direct=true,
   # base case of the recursion
   M1primitive = primitive_closure(L, M1)
 
-  #=
-  # the following is slower than computing the automorphism_group generators of
-  # M1primitive outright
-  gensOM1 = automorphism_group_generators(M1, depth = depth, bacher_depth = bacher_depth)
-  OM1 = matrix_group(gensOM1)
-  if M1primitive == M1
-    O1 = OM1
-  else
-    # M1 is generated by its shortest vectors only up to finite index
-    @vprint :Lattice 3 "Computing overlattice stabilizers \n"
-    O1,_ = stabilizer(OM1, M1primitive, on_lattices)
-  end
-  =#
-  O1 = matrix_group(automorphism_group_generators(M1primitive, depth = depth, bacher_depth = bacher_depth))
+  O1 = matrix_group(automorphism_group_generators(M1primitive; depth, bacher_depth))
   _set_nice_monomorphism!(O1, sv1; closed)
```

Comment:
```suggestion
  # _set_nice_monomorphism!(O1, sv1; closed)
```
These nice monomorphisms should be disabled. It seems that they make everyhing slower by a factor of 10. 
Can you confirm this?

Reply:
Probably our action function for them is too slow. One could work with Int vectors instead of fmpz etc to get it better.  But I don't know how much of a difference it makes in the end.

Reply:
OK, I will have a look at it. For me the biggest problems with kernel was that GAP did not know the domains were finite. I have never tried to see what happens if I remove the nice monomorphism part.

Reply:
Maybe @fingolfin  has an idea how we can make this action function fast? (matrices on vectors)
What we should do in any case is not to take all orbits of short vectors just enough orbits so that they generate a full rank sublattice.

Reply:
We have the possibility in Hecke to get a system a shortest vectors that does span the lattice spanned by shortest vectors. Considering what you said, would that be enough to consider such vectors and their orbits ?

Reply:
`function _shortest_vectors_sublattice_gram_integral(G::ZZMatrix)`

Reply:
What I mean is the following:
What we do at the moment is to compute the set `S` of all short vectors of L whose squared length is in a given intervall and let O(L) act on it to get a permutation representation. What we want is a monomorphism O(L) -> Perm(S).  To get such a monomorphism the following two properties are sufficient: 1. `S` gets permuted by `O(L)`. 2. `Span_QQ(S) = QQL` 
The following would be an algorithm to achieve this: take a single shortest vector, form its orbit and see if that spans(L). If not, take the orthogonal complement of it in `L` and add the orbit of its shortest vector and so on. 

But this is kind of tangential and will help only in certain cases. But not in others. 
What we still need is a faster action function.